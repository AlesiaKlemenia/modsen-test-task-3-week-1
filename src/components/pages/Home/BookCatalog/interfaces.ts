export interface ICatalogProps {
  children: JSX.Element[] | JSX.Element;
}
