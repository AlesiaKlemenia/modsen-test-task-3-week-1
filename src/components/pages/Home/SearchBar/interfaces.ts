import { IFilterBarProps } from '@/interfaces/IFilterBarProps';
import { ISearchProps } from '@/interfaces/ISearchProps';

export interface ISearchBarProps extends ISearchProps, IFilterBarProps {}
