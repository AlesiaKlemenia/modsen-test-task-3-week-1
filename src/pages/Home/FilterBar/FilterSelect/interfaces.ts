import { SelectChangeEvent } from '@mui/material';

import { ISelectItem } from '@/interfaces/ISelectItem';

export interface IFilterSelectProps {
  label: string;
  items: ISelectItem[];
  value: string;
  onSelectChange: (e: SelectChangeEvent<unknown>) => void;
}
