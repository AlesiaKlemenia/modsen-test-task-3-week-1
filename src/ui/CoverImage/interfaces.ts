export interface ICoverImageProps {
  title: string;
  coverUrl: string;
}
