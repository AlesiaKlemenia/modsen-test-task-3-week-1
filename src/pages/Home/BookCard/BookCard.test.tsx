import '@testing-library/jest-dom';

import { StyledEngineProvider } from '@mui/material';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';

import { IBookCardInfo } from '@/interfaces/IBookCardInfo';
import BookCard from '@/pages/Home/BookCard';

describe('BookCard component', () => {
  jest.mock('react-router-dom', () => ({
    Link: 'Link',
  }));

  it('renders correctly with full data', () => {
    const bookCardProps = {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [
        'Juvenile Fiction / Fantasy & Magic',
        'Fiction / Fantasy / General',
      ],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    };

    const componentRenderer = renderer.create(
      <StyledEngineProvider injectFirst>
        <MemoryRouter>
          <BookCard
            id={bookCardProps.id}
            title={bookCardProps.title}
            bookCategories={bookCardProps.bookCategories}
            authors={bookCardProps.authors}
            coverUrl={bookCardProps.coverUrl}
          />
        </MemoryRouter>
        ,
      </StyledEngineProvider>,
    );
    const tree = componentRenderer.toJSON();

    expect(tree).toMatchSnapshot();
  });

  const cases = [
    {
      id: 'QzHDDwAAQBAJ',
      title: '',
      bookCategories: [
        'Juvenile Fiction / Fantasy & Magic',
        'Fiction / Fantasy / General',
      ],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [
        'Juvenile Fiction / Fantasy & Magic',
        'Fiction / Fantasy / General',
      ],
      authors: [],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [
        'Juvenile Fiction / Fantasy & Magic',
        'Fiction / Fantasy / General',
      ],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl: '',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: '',
      bookCategories: [],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: '',
      bookCategories: [
        'Juvenile Fiction / Fantasy & Magic',
        'Fiction / Fantasy / General',
      ],
      authors: [],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: '',
      bookCategories: [
        'Juvenile Fiction / Fantasy & Magic',
        'Fiction / Fantasy / General',
      ],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl: '',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [],
      authors: [],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl: '',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [
        'Juvenile Fiction / Fantasy & Magic',
        'Fiction / Fantasy / General',
      ],
      authors: [],
      coverUrl: '',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: '',
      bookCategories: [],
      authors: [],
      coverUrl:
        'http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: '',
      bookCategories: [],
      authors: ['J.K. Rowling', 'John Tiffany', 'Jack Thorne'],
      coverUrl: '',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: `Harry Potter und das verwunschene Kind`,
      bookCategories: [],
      authors: [],
      coverUrl: '',
    },
    {
      id: 'QzHDDwAAQBAJ',
      title: '',
      bookCategories: [],
      authors: [],
      coverUrl: '',
    },
  ];

  it.each(cases)(
    'renders correctly with not full data',
    ({ id, title, bookCategories, authors, coverUrl }: IBookCardInfo) => {
      const componentRendererWithoutBookCategories = renderer.create(
        <StyledEngineProvider injectFirst>
          <MemoryRouter>
            <BookCard
              id={id}
              title={title}
              bookCategories={bookCategories}
              authors={authors}
              coverUrl={coverUrl}
            />
          </MemoryRouter>
          ,
        </StyledEngineProvider>,
      );
      const treeWithoutBookCategories =
        componentRendererWithoutBookCategories.toJSON();

      expect(treeWithoutBookCategories).toMatchSnapshot();
    },
  );
});
