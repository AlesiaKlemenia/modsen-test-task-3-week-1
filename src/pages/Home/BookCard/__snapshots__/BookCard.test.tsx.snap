// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard component renders correctly with full data 1`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        >
          Juvenile Fiction / Fantasy & Magic
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 1`] = `
[
  <a
    className="sc-dmyCSP eVTamN"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={true}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt=""
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        >
          Juvenile Fiction / Fantasy & Magic
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 2`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 3`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        >
          Juvenile Fiction / Fantasy & Magic
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          undefined, undefined
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 4`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src={null}
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        >
          Juvenile Fiction / Fantasy & Magic
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 5`] = `
[
  <a
    className="sc-dmyCSP eVTamN"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={true}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt=""
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 6`] = `
[
  <a
    className="sc-dmyCSP eVTamN"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={true}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt=""
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        >
          Juvenile Fiction / Fantasy & Magic
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          undefined, undefined
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 7`] = `
[
  <a
    className="sc-dmyCSP eVTamN"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={true}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt=""
          className="MuiBox-root css-i0loyx"
          src={null}
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        >
          Juvenile Fiction / Fantasy & Magic
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 8`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          undefined, undefined
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 9`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src={null}
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 10`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src={null}
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        >
          Juvenile Fiction / Fantasy & Magic
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          undefined, undefined
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 11`] = `
[
  <a
    className="sc-dmyCSP eVTamN"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={true}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt=""
          className="MuiBox-root css-i0loyx"
          src="http://books.google.com/books/publisher/content?id=QzHDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE71AmI8TP3JEaV01DOKOF1T6NA531p_djNucrd6UTjfx0XxtSaID1Eludk7BvwGVf_0DDM379yP43fWY0xKHknDCv1vWKTjIAFM6MDeQ9RGs83b0nKdsjvswitqdD1uQL2zk2Pjg&source=gbs_api"
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          undefined, undefined
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 12`] = `
[
  <a
    className="sc-dmyCSP eVTamN"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={true}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt=""
          className="MuiBox-root css-i0loyx"
          src={null}
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          J.K. Rowling, John Tiffany
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 13`] = `
[
  <a
    className="sc-dmyCSP hhSEDr"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt="Harry Potter und das verwunschene Kind"
          className="MuiBox-root css-i0loyx"
          src={null}
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        >
          Harry Potter und das verwunschene Kind
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          undefined, undefined
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;

exports[`BookCard component renders correctly with not full data 14`] = `
[
  <a
    className="sc-dmyCSP eVTamN"
    href="/books/QzHDDwAAQBAJ"
    isdisabled={true}
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-hLQSwg hvwzqE css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="sc-eDLKkx QPgvX"
      >
        <img
          alt=""
          className="MuiBox-root css-i0loyx"
          src={null}
        />
      </div>
      <div
        className="sc-jTQCzO nssjt"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 sc-fsYfdN eNcHzc css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-guDLey sc-gLLuof kTlak hlgxde css-ahj2mt-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 sc-beySPh sc-iBdnpw immIe cfpYwg css-ahj2mt-MuiTypography-root"
        >
          undefined, undefined
        </p>
      </div>
    </div>
  </a>,
  ",",
]
`;
